### DISCUSSION TITLE
How can I define Enum into request model?

### DISCUSSION THREAD

**Discussion by Hedgehogues** [fastapi/fastapi#7453] [Questions]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I define Enum into request model?

```
class Processors(enum.Enum):
    Python = 'python'


class PostRepository(BaseModel):
    file: str
    processor: Processors

```
I have such endpoint:

```
@app.post('/repositories', status_code=201)
async def post_repositories(request: PostRepository):
    pass
```

After starting application, FastAPI don't take Enum field:

![image](https://user-images.githubusercontent.com/4973667/83524221-f8672300-a4eb-11ea-933e-9b0d6e107a65.png)

**Comment by dbanty** [comment]
The SwaggerUI I find often has trouble displaying Enums. Try the redoc docs at `/redoc` instead of `/docs` to see if it's displayed.  You can also check the `/openapi.json` document directly to make sure your Enum is included.

You can still test the endpoint using your enum value, SwaggerUI just doesn't include it in the body for some reason. Might be fixed once Pydantic starts declaring enums as schemas in the next version? Not sure.

**Comment by zamiramir** [comment]
Could you try this?

```python
class Processors(str, enum.Enum):
    Python = 'python'


class PostRepository(BaseModel):
    file: str
    processor: Processors
```

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

I imagine you solved your problem, so thanks for closing the issue :+1:

**Comment by YuriiMotov** [comment]
Initial code works fine in current version (0.115.14)
![image](https://github.com/user-attachments/assets/608718cf-b588-43f7-9a6d-c5ade2071075)
